Frontend/src/App.svelte

```diff
@@ -120,63 +123,105 @@
     const res = await fetch(path);
-    if( res.status == 200 ){
-        let state = await res.json();
-        if(state.status === 'completed') {
-            updating = false;  // We got the last status.
-            lastStatus = '';
-            lastStepName = '';
-            if (state.conclusion === "success") {
-                logMessage( `Current build status: Success!`);
-            } else {
-
-                logMessage(`Current build status: Failed! (${state.stepNumber}/${state.stepCount}) - ${state.stepName}`);
-                logMessage(`Error: ${state.errorMessage}`);
-
-
-            }
-            return;
-        } else if (lastStatus === state.status && lastStepName === state.stepName) {
-            // Status did not change, not logging.
-            return;
-        } else if (state.status === 'in_progress') {
-            lastStatus = state.status;
-            lastStepName= state.stepName;
-            logMessage( `Current build status: In Progress (${state.stepNumber}/${state.stepCount}) - ${state.stepName}`);
-            return;
+    if (res.status == 200) {
+      let state = await res.json();
+      if (state.status === "completed") {
+        updating = false; // We got the last status.
+        lastStatus = "";
+        lastStepName = "";
+        if (state.conclusion === "success") {
+          logMessage(`Current build status: Success!`, "lime");
         } else {
-            lastStatus = state.status;
-            lastStepName= state.stepName;
-            logMessage( `Current build status: ${state.status}`);
-            return;
+          logMessage(
+            `Current build status: Failed! (${state.stepNumber}/${state.stepCount}) - ${state.stepName}`,
+            "salmon"
+          );
+          logMessage(`Error: ${state.errorMessage}`, "salmon");
         }
+        return;
+      } else if (
+        lastStatus === state.status &&
+        lastStepName === state.stepName
+      ) {
+        // Status did not change, not logging.
+        return;
+      } else if (state.status === "in_progress") {
+        lastStatus = state.status;
+        lastStepName = state.stepName;
+        logMessage(
+          `Current build status: In Progress (${state.stepNumber}/${state.stepCount}) - ${state.stepName}`,
+          "turquoise"
+        );
+        return;
+      } else {
+        lastStatus = state.status;
+        lastStepName = state.stepName;
+        logMessage(`Current build status: ${state.status}`, "turquoise");
+        return;
+      }
     }
   }
 
-  function logMessage(message) {
-    var newMessage = new Date(Date.now()).toISOString().substring(0, 23) + " - " + message + "\n";
-    log = log + newMessage;
+  async function logMessage(message, color = "white") {
+    message =
+      new Date(Date.now()).toISOString().substring(0, 23) +
+      " - " +
+      message +
+      "\n";
+
+    // format color string for style
+    if (color.substring(0, 6) != "color:") {
```

Comment:
Does this now mean that in line 195 object will read for example:
{
  message: "message here",
  color: "color:white"
}?

I see the use in line 253 and is ok.
Would it be more clear if the object would be like:
{
  message: "message here",
  **style**: "color:white"
}?
This would also make it make more sense if we in the future decide to use other style modifications.